packages/p2p-v2/src/pages/my-ads/components/AdPaymentDetailsSection/AdPaymentDetailsSection.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { MouseEventHandler, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { BUY_SELL } from '@/constants';
+import { AdFormController } from '../AdFormController';
+import { AdPaymentSelection } from '../AdPaymentSelection';
+import { AdSummary } from '../AdSummary';
+import { OrderTimeSelection } from '../OrderTimeSelection';
+
+type TAdPaymentDetailsSection = {
+    currency: string;
+    getCurrentStep: () => number;
+    getTotalSteps: () => number;
+    goToNextStep: MouseEventHandler<HTMLButtonElement>;
+    goToPreviousStep: MouseEventHandler<HTMLButtonElement>;
+    localCurrency?: string;
+    rateType: string;
+};
+
+const AdPaymentDetailsSection = ({ currency, localCurrency, rateType, ...props }: TAdPaymentDetailsSection) => {
+    const {
+        formState: { errors, isValid },
+        getValues,
+        setValue,
+    } = useFormContext();
+
+    const [selectedPaymentMethods, setSelectedPaymentMethods] = useState<(number | string)[]>(
+        getValues('payment-method') ?? []
+    );
+    const isSellAdvert = getValues('ad-type') === BUY_SELL.SELL;
+
+    const onSelectPaymentMethod = (paymentMethod: number | string) => {
+        if (selectedPaymentMethods.includes(paymentMethod)) {
+            const newSelectedPaymentMethods = selectedPaymentMethods.filter(method => method !== paymentMethod);
+            setSelectedPaymentMethods(selectedPaymentMethods.filter(method => method !== paymentMethod));
```

Comment:
would this work?

``suggestion
            setSelectedPaymentMethods(newSelectedPaymentMethods);
```